README.md

```diff
@@ -32,6 +36,104 @@ Most importantly, you will need to provide:
+
+The configuration file (different from `.env` file) is currently generated via a Google Spreadsheet manually curated by Starling Lab, which specifies how each `organization:collection` is assigned a set of processing `actions` alongside processing `params`. An example of these files is included [here](./config.example.json). Each time this file changes, the server must be restarted for internal states to update.
+
+A ZIP file entering the Integrity Backend, called `inputBundle`, is dropped into a folder associated with a particular `organization:collection`, and must adhere to a specific standard. It is a ZIP file containing three files, named as follow:
+
+```
+sha256(inputBundle).zip
+    sha256(content).ext
+    sha256(content)-meta-content.json
+    sha256(content)-meta-recorder.json
+```
+
+It is important that the ZIP file itself is named `sha256(inputBundle)`, as that ensures any change in the file content will be treated as a new asset by the data pipeline. This standardized naming is validated by the pipeline.
+
+Examples of the `meta-content` and `meta-recorder` files are included [here](https://github.com/starlinglab/integrity-schema/tree/main/integrity-backend).
+
+After the `inputBundle` is accepted into the data processing queue, actions assigned to the `organization:collection` will trigger.
+
+### Actions
+
+See [actions.py](https://github.com/starlinglab/integrity-backend/blob/main/integritybackend/actions.py).
+
+#### `archive`
+
+The `archive` action signs and registers the `inputBundle` onto several L1s according to the following procedure. This adds authenticity data in the archived bundles, and leaves cryptographic hashes on public L1s to establish proof-of-existence records.
+
+Each file in the `inputBundle` is signed using [webrecorder/authsign](https://github.com/webrecorder/authsign), then registered on the Bitcoin network with [OpenTimestamps](https://opentimestamps.org). The proof files generated by these two steps are appended to the cloned `inputBundle`:
+
+```
+sha256(archive).zip
+    sha256(content).ext
+    sha256(content)-meta-content.json
+    sha256(content)-meta-recorder.json
+    proofs/
+        sha256(content).ext.authsign
+        sha256(content)-meta-content.json.authsign
+        sha256(content)-meta-recorder.json.authsign
+        sha256(content).ext.ots
+        sha256(content)-meta-content.json.ots
+        sha256(content)-meta-recorder.json.ots
+```
+
+The new ZIP generated is referred to as the _archive_. The archive is then encrypted using `aes-256-cbc` with a key that is generated per `organization:collection`. The encrypted file is referred to as the _encrypted archive_, and it is usually stored on centralized storage and/or decentralized storage networks. The provisioning of access (i.e. the availability of the file and sharing of the AES key) are not handled by the Integrity Backend.
```

Comment:
We use aes-256-cbc but the encryption is defined in the config file per collection.